api/traces/v1/http.go

```diff
@@ -163,6 +165,47 @@ func NewV2Handler(read *url.URL, opts ...HandlerOption) http.Handler {
 	return r
 }
 
+// Parse a URL; if the URL includes `{tenant}` that portion will be replaced by the tenant
+func ExpandTemplatedUpstream(templateUpstream, tenant string) (*url.URL, error) {
+	rawTracesReadEndpoint := strings.Replace(templateUpstream, "{tenant}", tenant, 1)
+	return url.ParseRequestURI(rawTracesReadEndpoint)
+}
+
+// middlewareSetTemplatedUpstream is a variation of proxy.MiddlewareSetUpstream()
+// with additional processing if the upstream includes "{tenant}"
+func middlewareSetUpstream(logger log.Logger, read *url.URL, readTemplate string) proxy.Middleware {
+	if read != nil {
```

Comment:
Yes.

From the point of view of Observatorium users this is not a problem; the flags are validated and Observatorium exists with "only one of --traces.read.endpoint and --experimental.traces.read.endpoint-template allowed" if the user supplied both.

If someone tried to import Observatorium-API as a library in another program it might be unclear that this function ignores the template if URL is supplied.  What do you think?

Reply:
Sorry, I'm still confused with the behavior of the 2 flags. If `-experimental.traces.read.endpoint-template` is provided but `--traces.read.endpoint` is provided too, `--traces.read.endpoint` will be used here? Wouldn't this be caught [here](https://github.com/observatorium/api/blob/099ad6d12754959be92647e620ebb62b55dc5d66/main.go#L1037-L1038) already?